### PR TITLE
switch support

### PR DESCRIPTION
**Pull request by sorenoid**
* the switch is passed to the OutlinedTextBox as the suffix parameter.
* refactored trailingIcon in BaseTextField to run in a separate composable function that returns a nullable composable function, which is null when no icon is displayed, as opposed to an empty composable. The logic was rather long so I factored it out, and an empty composable  uses space at the end of the OutlinedTextBox that bumps over things like the suffix composable.
* fallback logic to combo box
    * when the value is empty or is not in the codes of the on or off CodedValue properties of the SwitchInput
    * small digression from the design spec where it says `if the field's value is any other value that is neither yes/on/true nor no/off/false, the UI for this field is a combo-box with that other value selected`. The other value is not selected, because there is no CodedValue in the combo box list to provide this option. If the user makes no selection in the combo box, the original value will remain.

### CODE REVIEW

**Review by kaushikrw — APPROVED** [review]
Looks good! Just a minor comment.

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/codedvalue/CodedValueFieldState.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ internal open class CodedValueFieldState(
     /**
      * The list of coded values associated with this field.
      */
-    val codedValues: List<CodedValue> = properties.codedValues
+    open val codedValues: List<CodedValue> = properties.codedValues
```
[the remark this answers is not in the document]
Comment (sorenoid) [reply toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/codedvalue/CodedValueFieldState.kt:78]: yes, the list is created for the Switch from the on and off value CodedValue properties

[the remark this answers is not in the document]
Comment (kaushikrw) [reply toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/codedvalue/CodedValueFieldState.kt:78]: But the nothing is overriding the `codedValues` as far as I can tell, so why does it need to be marked `open`?

[the remark this answers is not in the document]
Comment (sorenoid) [reply toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/components/codedvalue/CodedValueFieldState.kt:78]: oh you are right. The plan was to provide an override but it just gets thrown together in the remember/saver functions. I'll remove it.

**Pull request merged**